Sources/RoutingKit/Router.swift

```diff
@@ -5,10 +5,13 @@ public protocol Router {
     /// Type of value stored in routes. This will be returned by the router.
     associatedtype Output
     
-    /// Registers a new `Route` to the `Router`.
+    /// Registers a new `Output` to the `Router` at a given path.
     ///
-    /// Extraneous information such as `userInfo` may be discarded.
-    mutating func register(route: Route<Output>)
+    ///  - parameters:
+    ///     - output: Output to register.
+    ///     - path: Path to register output at.
+    ///
+    mutating func register(_ output: Output, at path: [PathComponent])
```

Comment:
I think I'd prefer `register(path:output:)` here since it better aligns with the other method on this protocol `route(path:parameters:)`

Reply:
I'd be fine with that, but you register output at a path, right? Or would you then rather have `register(path, with: output)`.

Reply:
I think just `register(path:output:)`. It's very much a nit though, approving.